pandas/types/concat.py

```diff
@@ -201,6 +201,52 @@ def convert_categorical(x):
         return Categorical(concatted, rawcats)
 
 
+def union_categoricals(to_union):
+    """
+    Combine list-like of Categoricals, unioning categories. All
+    must have the same dtype, and none can be ordered.
+
+    .. versionadded 0.18.2
+
+    Parameters
+    ----------
+    to_union : list-like of Categoricals
+
+    Returns
+    -------
+    Categorical
+       A single array, categories will be ordered as they
+       appear in the list
+
+    Raises
+    ------
+    TypeError
+        If any of the categoricals are ordered or all do not
+        have the same dtype
+    """
+    from pandas import Index, Categorical
+
+    if any(c.ordered for c in to_union):
+        raise TypeError("Can only combine unordered Categoricals")
+
+    first = to_union[0]
+    if not all(com.is_dtype_equal(c.categories.dtype, first.categories.dtype)
+               for c in to_union):
+        raise TypeError("dtype of categories must be the same")
+
+    cats = first.categories
+    unique_cats = cats.append([c.categories for c in to_union[1:]]).unique()
```

Comment:
Actually `drop_duplicates` is quite a bit slower, it looks like it's coercing to object rather than using the hash tables that `unique` does.

```python
In [74]: %timeit Index(a).append(Index(b)).drop_duplicates()
10 loops, best of 3: 197 ms per loop

In [75]: %timeit Index(Index(a).append(Index(b)).unique())
10 loops, best of 3: 71.5 ms per loop
```